src/test/java/htsjdk/TestDataProviders.java

```diff
@@ -0,0 +1,75 @@
+package htsjdk;
+
+import htsjdk.utils.ClassFinder;
+import org.testng.Assert;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.NoInjection;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * This test is a mechanism to check that none of the data-providers fail to run.
+ * It is needed because in the case that a data-provider fails (for some reason, perhaps a change in some other code
+ * causes it to throw an exception) the tests that rely on it will be silently skipped.
+ * The only mention of this will be in test logs but since we normally avoid reading these logs and rely on the
+ * exit code, it will look like all the tests have passed.
+ *
+ * @author Yossi Farjoun
+ */
+public class TestDataProviders {
+
+
+    @Test
+    public void IndependentTestOfDataProviderTest() throws IllegalAccessException, InvocationTargetException, InstantiationException {
+        testAllDataProvidersdata();
+    }
+
+    @DataProvider(name = "DataprovidersThatDontTestThemselves")
+    public Iterator<Object[]> testAllDataProvidersdata() throws IllegalAccessException, InstantiationException, InvocationTargetException {
+
+        List<Object[]> data = new ArrayList<>();
+        final ClassFinder classFinder = new ClassFinder();
+        classFinder.find("htsjdk", Object.class);
```

Comment:
should we restrict this to subclasses of HtsjdkTest?

Reply:
Now that we have this in htjsjdk, maybe we should also test that all `@Test` annotations are in a subclass of `HtsjdkTest` since they will fail to run otherwise.  `MetaTestTest.java`

Reply:
that's a great idea! I just implemented and found many classes in violation! hurray!

Reply:
(including the one I just added!!!)

Reply:
Also added test for non public methods (found a few...)